ext/fetch/lib.rs

```diff
@@ -214,6 +212,58 @@ pub fn get_or_create_client_from_state(
   }
 }
 
+struct ResourceToBodyAdapter(Rc<dyn Resource>, Option<Pin<Box<dyn Future<Output = Result<BufView, Error>>>>>);
```

Comment:
This pattern is identical to one in ext/http, so it might be worth turning this into a deno_core utility.